### PR TITLE
add email preference feature

### PR DESCRIPTION
**Pull request by holotsvan**
[https://github.com/ita-social-projects/GreenCity/issues/7380](https://github.com/ita-social-projects/GreenCity/issues/7380)

- Add UserNotificationPreferences entity + update User entity
- Add EmailPreferences enum
- Update profile endpoint and resp req dtos for it
- Add tests

### CODE REVIEW

**Review by Maryna-511750 — APPROVED** [review]

**Review by IliaWithHat — APPROVED** [review]

File: service/src/main/java/greencity/service/UserServiceImpl.java [hunk 1]
```diff
@@ -37,12 +37,7 @@
 import greencity.dto.user.DeactivateUserRequestDto;
 import greencity.dto.user.UserWithOnlineStatusDto;
 import greencity.dto.user.UserLocationDto;
-import greencity.entity.Language;
-import greencity.entity.SocialNetwork;
-import greencity.entity.SocialNetworkImage;
-import greencity.entity.User;
-import greencity.entity.UserDeactivationReason;
-import greencity.entity.UserLocation;
+import greencity.entity.*;
```
Comment (KizerovDmitriy) [inline service/src/main/java/greencity/service/UserServiceImpl.java]: we don't use wildcard imports

File: service/src/main/java/greencity/service/UserServiceImpl.java [hunk 2]
```diff
@@ -61,17 +56,15 @@
...
 import java.sql.Timestamp;
 import java.time.LocalDateTime;
 import java.time.ZoneId;
 import java.time.ZonedDateTime;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
+import java.util.*;
```
Comment (KizerovDmitriy) [inline service/src/main/java/greencity/service/UserServiceImpl.java]: same

File: service/src/test/java/greencity/service/UserServiceImplTest.java [hunk 3]
```diff
@@ -64,10 +64,7 @@
 import greencity.exception.exceptions.WrongIdException;
 import greencity.exception.exceptions.UserDeactivationException;
 import greencity.filters.UserSpecification;
-import greencity.repository.LanguageRepo;
-import greencity.repository.UserDeactivationRepo;
-import greencity.repository.UserLocationRepo;
-import greencity.repository.UserRepo;
+import greencity.repository.*;
```
Comment (KizerovDmitriy) [inline service/src/test/java/greencity/service/UserServiceImplTest.java]: and here

**Review by KizerovDmitriy — CHANGES_REQUESTED** [review]

File: service/src/test/java/greencity/service/UserServiceImplTest.java (hunk 3 above)
Comment (holotsvan) [reply service/src/test/java/greencity/service/UserServiceImplTest.java]: remove wildcards

**Review by KizerovDmitriy — APPROVED** [review]

**Pull request merged**